corehq/apps/translations/app_translations/upload_app.py

```diff
@@ -64,24 +66,35 @@ def process_bulk_app_translation_upload(app, workbook, expected_headers, lang=No
     :return: Returns a list of message tuples. The first item in each tuple is
     a function like django.contrib.messages.error, and the second is a string.
     """
+
+    def get_expected_headers(sheet_name):
```

Comment:
This function does its best to return the headers we expect, based on the current app, for an uploaded sheet. If the sheet is old, it might not include the unique IDs of the modules/forms. In that case `sheet_name_to_unique_id` will be empty and we fall back to using the name of the sheet, and hope that modules/forms have not been moved since the sheet was originally downloaded.

If a user created a new sheet, or renamed a sheet, or a form/module has been deleted since this sheet was downloaded, then expected headers will not be found. We return an empty list, and `_check_for_sheet_error()` will handle it.